components/automate-cli/pkg/verifyserver/models/batchcheck.go

```diff
@@ -78,6 +81,191 @@ type Config struct {
+			c.ExternalOS.OSUsername = awsManagedServicesConfig.ManagedOpensearchUsername
+			c.ExternalOS.OSRoleArn = awsManagedServicesConfig.AwsOsSnapshotRoleArn
+		}
+	}
+
+	if haConfig.IsExistingInfra() {
+		existingInfraConfig := haConfig.ExistingInfra.Config
+		c.Hardware.AutomateNodeIps = existingInfraConfig.AutomatePrivateIps
+		c.Hardware.ChefInfraServerNodeIps = existingInfraConfig.ChefServerPrivateIps
+		automateConfig := haConfig.Automate.Config
+		if automateConfig.EnableCustomCerts {
+			c.appendCertsByIpToNodeCerts(automateConfig.CertsByIP, haConfig.ExistingInfra.Config.AutomatePrivateIps, automateConfig.PrivateKey, automateConfig.PublicKey, "", "")
+		}
+
+		chefserverConfig := haConfig.ChefServer.Config
+		if chefserverConfig.EnableCustomCerts {
+			c.appendCertsByIpToNodeCerts(chefserverConfig.CertsByIP, haConfig.ExistingInfra.Config.ChefServerPrivateIps, chefserverConfig.PrivateKey, chefserverConfig.PublicKey, "", "")
+		}
+
+		if !haConfig.IsExternalDb() {
+			c.Hardware.PostgresqlNodeIps = existingInfraConfig.PostgresqlPrivateIps
+			c.Hardware.OpenSearchNodeIps = existingInfraConfig.OpensearchPrivateIps
+			postgresqlConfig := haConfig.Postgresql.Config
+			if postgresqlConfig.EnableCustomCerts {
+				c.appendCertsByIpToNodeCerts(postgresqlConfig.CertsByIP, haConfig.ExistingInfra.Config.PostgresqlPrivateIps, postgresqlConfig.PrivateKey, postgresqlConfig.PublicKey, "", "")
+			}
+
+			openSearchConfig := haConfig.Opensearch.Config
+			if openSearchConfig.EnableCustomCerts {
+				c.appendCertsByIpToNodeCerts(openSearchConfig.CertsByIP, haConfig.ExistingInfra.Config.OpensearchPrivateIps, openSearchConfig.PrivateKey, openSearchConfig.PublicKey, openSearchConfig.AdminKey, openSearchConfig.AdminCert)
+			}
+		}
+
+		if haConfig.IsExternalDb() {
+
+			externalPgConfig := haConfig.External.Database.PostgreSQL
+			c.ExternalPG.PGDbUserName = externalPgConfig.DbuserUsername
+			c.ExternalPG.PGDbUserPassword = externalPgConfig.DbuserPassword
+			c.ExternalPG.PGInstanceURL = externalPgConfig.InstanceURL
+
+			// pg root-ca might be nil in pre deploy state
+			c.ExternalPG.PGRootCert = externalPgConfig.PostgresqlRootCert
+			c.ExternalPG.PGSuperuserName = externalPgConfig.SuperuserUsername
+			c.ExternalPG.PGSuperuserPassword = externalPgConfig.SuperuserPassword
+
+			externalOsConfig := haConfig.External.Database.OpenSearch
+			c.ExternalOS.OSDomainName = externalOsConfig.OpensearchDomainName
+			c.ExternalOS.OSDomainURL = externalOsConfig.OpensearchDomainURL
+
+			// os root-ca might be nil in pre deploy state
+			c.ExternalOS.OSCert = externalOsConfig.OpensearchRootCert
+			c.ExternalOS.OSUserPassword = externalOsConfig.OpensearchUserPassword
+			c.ExternalOS.OSUsername = externalOsConfig.OpensearchUsername
+
+			if haConfig.IsAwsExternalOsConfigured() {
+				c.ExternalOS.OSRoleArn = externalOsConfig.Aws.AwsOsSnapshotRoleArn
+			}
+		}
+	}
+
+	// not available in config
+	c.DeploymentState = ""
+	c.APIToken = ""
```

Comment:
We don't need API token now, as we are not calling any automate API

Reply:
Okey.